### PR TITLE
config: organize and provide doc links [more]

### LINKED ISSUE #19: Organize configurations
Per the discussion @ppanero and I had with @lnielsen : 

`invenio.cfg` should actually be user-friendly. This means it should be TIDY and WELL-DOCUMENTED. It obviously has to contain the cookiecutter dependent configuration variables.
However, anything else that is complicated/more advanced should go in `config.py` of `invenio-app-rdm` or respective modules. Anything "dirty". 
    
For example, REST endpoints should not be in `invenio.cfg`.

For this task, let's organize invenio.cfg . Django's starter [settings.py](https://github.com/django/django/blob/05d0eca635853564c57e639ac5590674a7de2ed6/django/conf/project_template/project_name/settings.py-tpl) was given as an example of user-friendly, well organized configuration file.

### PR DESCRIPTION
**Pull request by fenekku**
Only 'user-friendly' configuration is kept in invenio.cfg.
The configuration criteria to assess this friendliness are:
- not big
- not too low-level
- required or likely to be customized
- dependent on cookiecutter value

Closes #19

### CODE REVIEW

File: {{cookiecutter.project_shortname}}/invenio.cfg [hunk 1]
```diff
@@ -1,15 +1,54 @@
-ACCOUNTS_SESSION_REDIS_URL="redis://localhost:6379/1"
-BROKER_URL="amqp://guest:guest@localhost:5672/"
-CACHE_REDIS_URL="redis://localhost:6379/0"
-CACHE_TYPE="redis"
-CELERY_BROKER_URL="amqp://guest:guest@localhost:5672/"
-CELERY_RESULT_BACKEND="redis://localhost:6379/2"
-SEARCH_ELASTIC_HOSTS=[{"host": "localhost", "port": 9200}]
```
Comment (fenekku) [inline {{cookiecutter.project_shortname}}/invenio.cfg]: Removed ones are in invenio-app-rdm anyway.

Comment (ppanero) [reply {{cookiecutter.project_shortname}}/invenio.cfg]: Minor: I kind of consider the "endpoints" of this services are super likely to be customized, no?

Comment (fenekku) [reply {{cookiecutter.project_shortname}}/invenio.cfg]: Yes, that's another aspect we didn't talk about much: we still have to think about the dev vs real prod configuration story. If the hoster modifies the `invenio.cfg` for production, then running it locally (dev) will probably not work anymore (invalid service hosts for instance). There should be a clear story here which could be documented as:

- Modify `invenio.cfg` when it affects your production and development environment
- Modify `.production.env` when it affects only your production environment

and then we have invenio-cli place that `.production.env` at the right place. Or something similar. I am making an issue for that so we can deal with it separately.

Comment (fenekku) [reply {{cookiecutter.project_shortname}}/invenio.cfg]: Task: https://github.com/inveniosoftware/invenio-cli/issues/52

**Pull request merged**